### PR TITLE
Handle some possible null-references

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
MicrosoftTeams-image
I did what seemed like the simple fix for this null-ref, but if that's insufficient and we'll need to dig deeper let me know.

### CODE REVIEW

**Comment by 333fred** [comment]
We'll need a test for this. Sorry.

File: src/Compilers/CSharp/Portable/Compilation/AttributeSemanticModel.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ internal sealed class AttributeSemanticModel : MemberSemanticModel
         /// <summary>
         /// Creates an AttributeSemanticModel that allows asking semantic questions about an attribute node.
         /// </summary>
-        public static AttributeSemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, AttributeSyntax syntax, NamedTypeSymbol attributeType, AliasSymbol aliasOpt, Symbol? attributeTarget, Binder rootBinder, ImmutableDictionary<Symbol, Symbol> parentRemappedSymbolsOpt)
+        public static AttributeSemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, AttributeSyntax syntax, NamedTypeSymbol attributeType, AliasSymbol aliasOpt, Symbol? attributeTarget, Binder rootBinder, ImmutableDictionary<Symbol, Symbol>? parentRemappedSymbolsOpt)
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Compilation/AttributeSemanticModel.cs:38]:
```suggestion
        public static AttributeSemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, AttributeSyntax syntax, NamedTypeSymbol attributeType, AliasSymbol? aliasOpt, Symbol? attributeTarget, Binder rootBinder, ImmutableDictionary<Symbol, Symbol>? parentRemappedSymbolsOpt)
```

File: src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs [hunk 2]
```diff
@@ -1268,13 +1268,14 @@ private SynthesizedRecordConstructor TryGetSynthesizedRecordConstructor(RecordDe
...
-        private AttributeSemanticModel CreateModelForAttribute(Binder enclosingBinder, AttributeSyntax attribute, MemberSemanticModel containingModel)
+#nullable enable
+        private AttributeSemanticModel CreateModelForAttribute(Binder enclosingBinder, AttributeSyntax attribute, MemberSemanticModel? containingModel)
         {
             AliasSymbol aliasOpt;
             var attributeType = (NamedTypeSymbol)enclosingBinder.BindType(attribute.Name, BindingDiagnosticBag.Discarded, out aliasOpt).Type;
 
             // For attributes where a nameof could introduce some type parameters, we need to track the attribute target
-            Symbol attributeTarget = getAttributeTarget(attribute.Parent.Parent);
+            Symbol? attributeTarget = getAttributeTarget(attribute.Parent?.Parent);
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs:1278]: Adding these `?.`s everywhere we access `Parent` is likely going to invalidate some invariant somewhere. I think we need to get an understanding of the syntax that is causing a null reference before making more changes.

Comment (ryanbrandenburg) [reply src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs:1278]: Unfortunately I lost the repro, and the particular syntax that set this off wasn't clear in the first place. Given that I'll close this and try again in a more targeted fashion the next time I see it.

**Pull request closed without merging**